geonode/maps/qgis_server_views.py

```diff
@@ -36,13 +53,188 @@ class MapCreateView(CreateView):
     context_object_name = 'map'
 
     def get_context_data(self, **kwargs):
-        # list all required layers
-        layers = Layer.objects.all()
-        context = {
-            'create': True,
-            'layers': layers
-        }
-        return context
+        request = self.request
+        if 'access_token' in request.session:
+            access_token = request.session['access_token']
+        else:
+            access_token = None
+
+            DEFAULT_MAP_CONFIG, DEFAULT_BASE_LAYERS = default_map_config(request)
+
+        if request.method == 'GET' and 'copy' in request.GET:
+            """Prepare context data."""
+            request = self.request
+            mapid = self.request.GET['copy']
+
+            map_obj = _resolve_map(
+                request, mapid, 'base.view_resourcebase', _PERMISSION_MSG_VIEW)
+
+            snapshot = self.kwargs.get('snapshot')
+            if snapshot is None:
+                config = map_obj.viewer_json(request.user, access_token)
+            else:
+                config = snapshot_config(snapshot, map_obj, request.user,
+                                         access_token)
+            # list all required layers
+            layers = Layer.objects.all()
+            map_layers = MapLayer.objects.filter(
+                map_id=mapid).order_by('stack_order')
+            context = {
+                'config': json.dumps(config),
+                'create': False,
+                'layers': layers,
+                'map': map_obj,
+                'map_layers': map_layers,
+                'preview': getattr(
+                    settings,
+                    'LAYER_PREVIEW_LIBRARY',
+                    '')
+            }
+            return context
+        else:
+            if request.method == 'GET':
+                params = request.GET
+            elif request.method == 'POST':
+                params = request.POST
+            else:
+                return self.render_to_response(status=405)
+
+            if 'layer' in params:
+                bbox = None
+                map_obj = Map(projection=getattr(settings, 'DEFAULT_MAP_CRS',
+                                                 'EPSG:900913'))
```

Comment:
We usually pull left, from the `settings` parameter. Can we also avoid 900913 and use 3857 instead? Is-it stored 900913 in the DB?